common/stage_timer.go

```diff
@@ -0,0 +1,100 @@
+package common
+
+import (
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+// It may be helpful to generalize this utility to be used in other parts of the codebase. For a future PR, perhaps.
+
+// StageTimer encapsulates metrics to help track the time spent in each stage of the payload dispersal process.
+type StageTimer struct {
+	stageCount   *prometheus.GaugeVec
+	stageLatency *prometheus.SummaryVec
+}
+
+// SequenceProbe tracks the timing of a single sequence of operations. Multiple sequences can be tracked concurrently.
```

Comment:
> can be tracked concurrently

Presumably with this you mean "can be tracked concurrently *by the same StageTimer*"?

Reply:
Yes. Updated.

```
// SequenceProbe can be used to track the amount of time that a particular operation spends doing particular 
// sub-operations (i.e. stages). Multiple instances of a particular operation can be tracked concurrently by the same 
// StageTimer. For each operation, the StageTimer builds a SequenceProbe. Each SequenceProbe the lifecycle of a single 
// iteration of an operation.
```